internal/eth/sendtx.go

```diff
@@ -18,88 +21,164 @@ import (
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/crypto/kzg4844"
 	"github.com/ethereum/go-ethereum/ethclient"
-	"github.com/ethereum/go-ethereum/log"
 	"github.com/holiman/uint256"
-	bb "github.com/primev/preconf_blob_bidder/core/mevcommit"
+	bb "github.com/primev/preconf_blob_bidder/internal/mevcommit"
+	"github.com/rs/zerolog/log"
 	"golang.org/x/exp/rand"
 )
 
+var defaultTimeout time.Duration
+
+// init initializes the defaultTimeout variable by reading the DEFAULT_TIMEOUT
+// environment variable. If not set or invalid, it defaults to 15 seconds.
+func init() {
```

Comment:
This is unnecessary, it should be set to it's default value where the flag is defined.

Reply:
The default timeout flag is defined in the main.go() file as a .env variable. The defaultTimeout variable is also used in the bundle.go file as well as sendtx.go.

This was the best I could come up with to use the flag defined value without refactoring the functions to require a defaulttimeout parameter

Reply:
I would suggest creating a service struct in which you set all the necessary options that are known at the time of creation and will not change (e.g. default timeout, rpcurl, etc.) and fields that must maintain some state (e.g. logger, ethclient, httpclient, etc.). Then attach existing functions to this structure. This will result in fewer parameters in these methods and avoid global state. Your code will also be easier to mock and test.